modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/EmailAddressResourceImpl.java

```diff
@@ -120,6 +163,64 @@ public Page<EmailAddress> getUserAccountEmailAddressesPage(
 				EmailAddressUtil::toEmailAddress));
 	}
 
+	@Override
+	public EmailAddress patchEmailAddress(
+			Long emailAddressId, EmailAddress emailAddress)
+		throws Exception {
+
+		com.liferay.portal.kernel.model.EmailAddress serviceEmailAddress =
+			_emailAddressService.getEmailAddress(emailAddressId);
+
+		return EmailAddressUtil.toEmailAddress(
+			_updateEmailAddress(serviceEmailAddress, emailAddress));
+	}
+
+	@Override
+	public EmailAddress patchEmailAddressByExternalReferenceCode(
+			String externalReferenceCode, EmailAddress emailAddress)
+		throws Exception {
+
+		com.liferay.portal.kernel.model.EmailAddress serviceEmailAddress =
+			_emailAddressService.fetchEmailAddressByExternalReferenceCode(
+				externalReferenceCode, contextCompany.getCompanyId());
+
+		if (serviceEmailAddress == null) {
+			throw new NoSuchEmailAddressException();
+		}
+
+		return EmailAddressUtil.toEmailAddress(
+			_updateEmailAddress(serviceEmailAddress, emailAddress));
+	}
+
+	private long _getListTypeId(String listTypeId) {
+		List<ListType> listTypes = _listTypeService.getListTypes(
+			contextCompany.getCompanyId(), listTypeId);
+
+		ListType listType = listTypes.get(0);
+
+		return listType.getListTypeId();
+	}
```

Comment:
I think it's wrong, see my comment below
(also because here from the id you're getting the id? (or if this listTypeId in reality is the name it's wrong the name of the variable and the fact that you're taking always the 0 (you should get the one with this name, for this company, with the same type of the className in the serviceEmailAddress object